lib/amazon.js

```diff
@@ -71,9 +71,10 @@ function updateRuntime ({ functionName }) {
   return lambda.updateFunctionConfiguration(params).promise().then(res => res.FunctionArn)
 }
 
-exports.updateFunction = async function ({ code, functionName }) {
+exports.updateFunction = async function ({ code, functionName, handler }) {
   const params = {
     FunctionName: functionName,
+    Handler: handler,
```

Comment:
Handler is not available to updateFunction, it needs to be set with `updateFunctionConfiguration`, see how Runtime is updated on line 84:

```js
   if (result.Runtime !== defaultParams.Runtime || result.Handler !== handler) {
     return updateRuntime({ functionName, handler })
   }